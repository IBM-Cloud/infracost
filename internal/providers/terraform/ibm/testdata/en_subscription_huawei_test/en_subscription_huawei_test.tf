terraform {
  required_providers {
    ibm = {
      source = "IBM-Cloud/ibm"
    }
  }
}

locals {
  event_notifications = {
    plans = ["lite", "standard"],
  }
}

resource "ibm_resource_instance" "event_notifications" {
  for_each          = toset(local.event_notifications.plans)
  name              = "event-notifications-${each.value}"
  location          = "us-south"
  plan              = each.value
  resource_group_id = "default"
  service           = "event-notifications"
}

resource "ibm_en_subscription_huawei" "subscription_huawei_lite" {
  instance_guid  = ibm_resource_instance.event_notifications["lite"].guid
  name           = "Huawei Subscription"
  description    = "Huawei Subscription for Event Notifications"
  destination_id = "somedestinationid"
  topic_id       = "sometopicid"
}

resource "ibm_en_subscription_huawei" "subscription_huawei_standard" {
  instance_guid  = ibm_resource_instance.event_notifications["standard"].guid
  name           = "Huawei Subscription"
  description    = "Huawei Subscription for Event Notifications"
  destination_id = "somedestinationid"
  topic_id       = "sometopicid"
}

resource "ibm_en_subscription_huawei" "subscription_huawei_standard_no_usage" {
  instance_guid  = ibm_resource_instance.event_notifications["standard"].guid
  name           = "Huawei Subscription"
  description    = "Huawei Subscription for Event Notifications"
  destination_id = "somedestinationid"
  topic_id       = "sometopicid"
}
