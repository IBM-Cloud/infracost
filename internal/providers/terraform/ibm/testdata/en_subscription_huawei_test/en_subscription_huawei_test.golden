
 Name                                                                             Monthly Qty  Unit                    Monthly Cost 
                                                                                                                                    
 ibm_en_subscription_huawei.subscription_huawei_lite                                                                                
 └─ Outbound Huawei Push Messages (Lite plan) (Max. 1,000 per destination)                  1  Messages                       $0.00 
                                                                                                                                    
 ibm_en_subscription_huawei.subscription_huawei_standard                                                                            
 └─ Outbound Huawei Push Messages (Standard plan) (first 100 Messages)                    100  Messages                       $0.00 
 └─ Outbound Huawei Push Messages (Standard plan) (over 100 Messages)                  10,000  Messages                       $0.01 
                                                                                                                                    
 ibm_en_subscription_huawei.subscription_huawei_standard_no_usage                                                                   
 └─ Outbound Huawei Push Messages (Standard plan) (first 100 Messages)      Monthly cost depends on usage: $0.000001 per Messages   
 └─ Outbound Huawei Push Messages (Standard plan) (over 100 Messages)       Monthly cost depends on usage: $0.0000005 per Messages  
                                                                                                                                    
 ibm_resource_instance.event_notifications["lite"]                                                                                  
 └─ Lite Plan                                                                               1  Instance                       $0.00 
                                                                                                                                    
 ibm_resource_instance.event_notifications["standard"]                                                                              
 └─ Ingested Events                                                         Monthly cost depends on usage: $1.08 per Million Events 
                                                                                                                                    
 OVERALL TOTAL                                                                                                                $0.01 
──────────────────────────────────
5 cloud resources were detected:
∙ 5 were estimated, 2 of which include usage-based costs, see https://infracost.io/usage-file